packages/core/src/notification.ts

```diff
@@ -0,0 +1,581 @@
+  const defaultStatus = notification.severity === notificationSeverityLevel.WARNING ? unknownGqlStatus.WARNING : unknownGqlStatus.INFORMATION
+  return new GqlStatusObject({
+    gql_status: defaultStatus.gql_status,
+    status_description: notification.description ?? defaultStatus.status_description,
+    neo4j_code: notification.code,
+    title: notification.title,
+    diagnostic_record: {
+      ...rawPolyfilledDiagnosticRecord,
+      _severity: notification.severity,
+      _classification: notification.category,
+      _position: notification.position
+    }
+  })
+}
+
+const rawPolyfilledDiagnosticRecord = {
+  OPERATION: '',
+  OPERATION_CODE: '0',
+  CURRENT_SCHEMA: '/'
+}
+
+Object.freeze(rawPolyfilledDiagnosticRecord)
+
+/**
+ * This objects are used for polyfilling the first status on the status list
+ *
+ * @private
+ */
+const staticGqlStatusObjects = {
+  SUCCESS: new GqlStatusObject({
+    gql_status: '00000',
+    status_description: 'note: successful completion',
+    diagnostic_record: rawPolyfilledDiagnosticRecord
+  }),
+  NO_DATA: new GqlStatusObject({
+    gql_status: '02000',
+    status_description: 'note: no data',
+    diagnostic_record: rawPolyfilledDiagnosticRecord
+  }),
+  NO_DATA_UNKNOWN_SUBCONDITION: new GqlStatusObject({
+    ...unknownGqlStatus.NO_DATA,
+    diagnostic_record: rawPolyfilledDiagnosticRecord
+  }),
+  OMITTED_RESULT: new GqlStatusObject({
+    gql_status: '00001',
+    status_description: 'note: successful completion - omitted result',
+    diagnostic_record: rawPolyfilledDiagnosticRecord
+  })
+}
+
+Object.freeze(staticGqlStatusObjects)
+
+/**
+ *
+ * @private
+ * @param metadata
+ * @returns
+ */
+function buildGqlStatusObjectFromMetadata (metadata: any): [GqlStatusObject, ...GqlStatusObject[]] {
+  function getGqlStatusObjectFromStreamSummary (summary: any): GqlStatusObject {
+    if (summary?.have_records_streamed === true) {
+      return staticGqlStatusObjects.SUCCESS
+    }
+
+    if (summary?.has_keys === false) {
+      return staticGqlStatusObjects.OMITTED_RESULT
+    }
+
+    if (summary?.pulled === true) {
+      return staticGqlStatusObjects.NO_DATA
+    }
+
+    return staticGqlStatusObjects.NO_DATA_UNKNOWN_SUBCONDITION
+  }
+
+  if (metadata.statuses != null) {
+    return metadata.statuses.map((status: unknown) => new GqlStatusObject(status))
+  }
+
+  const clientGenerated = getGqlStatusObjectFromStreamSummary(metadata.stream_summary)
+  const polyfilledObjects = [clientGenerated, ...(metadata.notifications?.map(polyfillGqlStatusObject) ?? []) as GqlStatusObject[]]
+
+  return polyfilledObjects.sort((a: GqlStatusObject, b: GqlStatusObject) => calculateWeight(a) - calculateWeight(b)) as [GqlStatusObject, ...GqlStatusObject[]]
+}
+
+const gqlStatusWeightByClass = Object.freeze({
+  '02': 0,
+  '01': 1,
+  '00': 2
+})
+/**
+ * GqlStatus weight
+ *
+ * @private
+ */
+function calculateWeight (gqlStatusObject: GqlStatusObject): number {
```

Comment:
Is the weight stuff just to ensure the status object is first in the array? I didn't see any weight mentioned in the ADR.